domains/authorization/Authorization.API/API/Authorization/[Features]/GetConsentForClientQuery.cs

```diff
@@ -22,21 +24,21 @@ public GetConsentForClientQueryHandler(IClientRepository clientRepository)
     public async Task<GetConsentForClientQueryResult> Handle(GetConsentForClientQuery query,
         CancellationToken cancellationToken)
     {
-        var idpClientID = new IdpClientId(query.ClientId);
+        var requestedClientId = new IdpClientId(query.ClientId);
 
-        var result = await _clientRepository.Query()
-            .Where(x => x.IdpClientId == idpClientID)
+        var foundClient = await _clientRepository.Query()
+            .Where(x => x.IdpClientId == requestedClientId)
             .Select(x => new GetConsentForClientQueryResult(query.ClientId, x.ClientType.ToString(), "someOrgName",
                 new[] { Guid.NewGuid(), Guid.NewGuid(), Guid.NewGuid() },
                 "dashboard production meters certificates wallet"))
-            .FirstAsync(cancellationToken);
+            .FirstOrDefaultAsync(cancellationToken);
 
-        if (result.OrgName == "Energinet")
+        if (foundClient is null)
         {
-            throw new Exception("Nooooooh!");
+            throw new EntityNotFoundException(query.ClientId, typeof(Client));
```

Comment:
Is this an exception that will be shown to the client? If yes then we usually dont return exceptions, on not found entities. We just return empty collection.

Reply:
This will result in a HTTP 404 not found